src/NServiceBus.Core/Persistence/Msmq/SubscriptionStorage/MsmqSubscriptionStorage.cs

```diff
@@ -55,8 +56,9 @@ public void Init()
             }
         }
 
-        public IEnumerable<string> GetSubscriberAddressesForMessage(IEnumerable<MessageType> messageTypes)
+        public Task<IEnumerable<string>> GetSubscriberAddressesForMessage(IEnumerable<MessageType> messageTypes)
         {
+            var subscribers = new List<string>();
             var result = new List<string>();
             var messagelist = messageTypes.ToList();
             lock (locker)
```

Comment:
Doesn't seem like a big issue. That said we need better perf tests for this in the future